app/forms/candidate_interface/other_qualification_type_form.rb

```diff
@@ -25,7 +32,12 @@ class OtherQualificationTypeForm
     def initialize(current_application = nil, intermediate_data_service = nil, options = nil)
       @current_application = current_application
       @intermediate_data_service = intermediate_data_service
-      options = @intermediate_data_service.read.merge(options.select { |_, value| value.present? }) if @intermediate_data_service
+
+      options = if @intermediate_data_service
+                  prepared_options = prepare(options)
+                  @intermediate_data_service.read.merge(prepared_options)
+                end
```

Comment:
this will reset `options` to nil if `@intermediate_data_service` is nil. Is that intended?

Reply:
Yep this is an error on my part - have fixed